### PR TITLE
Scripts/Wandering isle: Handle quest 29414 (The way of the Tushui)

### PR DESCRIPTION
**Pull request by aquadeus**
**Changes proposed:**

-  Implement quest 29414 more blizzlike

**Tests performed:**

Tested in-game

**Known issues and TODO list:** (add/remove lines as needed)

- Adjust spawn position of the npcs spawned by spell 114710 / 114698 (must be spawned outside the cave)

### CODE REVIEW

File: src/server/scripts/Pandaria/zone_the_wandering_isle.cpp [hunk 1]
```diff
@@ -400,10 +402,367 @@ struct npc_instructor_zhi : public ScriptedAI
...
+{
+    // Texts
+    SAY_GO_CAVE              = 0,
+
+    // Waypoints
+    POINT_JUMP               = 1,
+    PATH_CAVE_OF_MEDITATION  = 5965200
+};
+
+Position const aysaJumpPos[3] =
```
Comment (Shauren) [inline src/server/scripts/Pandaria/zone_the_wandering_isle.cpp:416]: can be constexpr now

File: src/server/scripts/Pandaria/zone_the_wandering_isle.cpp [hunk 2]
```diff
@@ -400,10 +402,367 @@ struct npc_instructor_zhi : public ScriptedAI
...
+    npc_aysa_cloudsinger_summon(Creature* creature) : ScriptedAI(creature) { }
+
+    void IsSummonedBy(WorldObject* summoner) override
+    {
+        if (!summoner->IsPlayer())
+            return;
+
+        Talk(SAY_GO_CAVE, summoner);
+
+        _scheduler.Schedule(Seconds(3), [this](TaskContext task)
```
Comment (Shauren) [inline src/server/scripts/Pandaria/zone_the_wandering_isle.cpp:435]: `Seconds(3)` -> `3s` (apply everywhere, too many to list)

File: src/server/scripts/Pandaria/zone_the_wandering_isle.cpp [hunk 3]
```diff
@@ -400,10 +402,367 @@ struct npc_instructor_zhi : public ScriptedAI
...
+                player->CastSpell(player, SPELL_MEDITATION_TIMER_BAR, TRIGGERED_FULL_MASK);
+
+            return true;
+        }
+        return false;
+    }
+
+    bool OnExit(Player* player, AreaTriggerEntry const* /*trigger*/) override
+    {
+        if (player->HasAura(SPELL_MEDITATION_TIMER_BAR))
```
Comment (Shauren) [inline src/server/scripts/Pandaria/zone_the_wandering_isle.cpp:519]: `HasAura` unneccessary before `RemoveAura`

File: src/server/scripts/Pandaria/zone_the_wandering_isle.cpp [hunk 4]
```diff
@@ -400,10 +402,367 @@ struct npc_instructor_zhi : public ScriptedAI
...
+                aysa->AI()->Talk(SAY_AYSA_FINISH_MEDITATION);
+                talkDone = true;
+            }
+        }
+
+        return true;
+    }
+
+private:
+    bool talkDone = false;
```
Comment (Shauren) [inline src/server/scripts/Pandaria/zone_the_wandering_isle.cpp:573]: Fundamentally broken, there is only one copy of `at_inside_of_cave_of_meditation` in worldserver, you will need to store it in script of the summoned creature

File: src/server/scripts/Pandaria/zone_the_wandering_isle.cpp [hunk 5]
```diff
@@ -400,10 +402,367 @@ struct npc_instructor_zhi : public ScriptedAI
...
+    {
+        if (Unit* target = GetTarget())
+        {
+            target->ModifyPower(POWER_ALTERNATE_POWER, 1);
+
+            if (target->GetPowerPct(POWER_ALTERNATE_POWER) == 100)
+            {
+                target->CastSpell(GetTarget(), SPELL_CAVE_OF_SCROLLS_CREDIT, TRIGGERED_FULL_MASK);
+                target->CastSpell(GetTarget(), SPELL_AYSA_CAVE_OF_SCROLLS_COMP, TRIGGERED_FULL_MASK);
+                target->RemoveAura(GetId());
```
Comment (Shauren) [inline src/server/scripts/Pandaria/zone_the_wandering_isle.cpp:744]: Just call `this->Remove()` (without this->)

**Review by Shauren — CHANGES_REQUESTED** [review]

**Review by Shauren — CHANGES_REQUESTED** [review]

**Comment by Aokromes** [comment]
Conflicting files
src/server/scripts/Pandaria/zone_the_wandering_isle.cpp

**Comment by aquadeus** [comment]
> Conflicting files src/server/scripts/Pandaria/zone_the_wandering_isle.cpp

done

**Pull request merged**